service/history/workflow/mutable_state_impl.go

```diff
@@ -5061,6 +5061,16 @@ func (ms *MutableStateImpl) RetryActivity(
 
 	// if activity is paused
 	if ai.Paused {
+		// need to update activity
```

Comment:
if activity is paused we don't want to execute all the code below.
It may fail, and it shouldn't fail. Example - it may be out of attempts, but if activity is paused that shouldn't lead to failure. Users can call UpdateActivity, increase the number of attempts, then unpause.